### ISSUE TITLE
Server side rendering always renders the homepage

### ISSUE THREAD

**Issue by jhades** [angular/universal-starter#79]
With versions  angular2": "2.0.0-beta.15, angular2-universal": "~0.95.1" of the starter (I just freshly cloned it now), I've noticed the following behaviour:

If we open the starter and go to the about page in [localhost:3000/about](localhost:3000/about]), and then hit refresh and inspect the response:

We notice that the home page is being rendered by the server, and not the about page.  We can see that in this  [video demo](https://s3-us-west-1.amazonaws.com/blog-jhades/Issue-routing-not-used-on-server-side.mp4).

Before this used to work OK, meaning the about page was rendered server side as expected. With the version of the starter that used Angular 2 Beta 14 this was working OK, it must have been a very recent commit.

To summarize the versions: 

- using the starter with universal 0.91.0 and angular 2 beta 14 - it works
- after upgrade to universal 0.95.1and angular 2 beta 14 - issue occurs

Note: The about page ends up getting displayed in the end, although it was not rendered on the server side.

This is due to some other issue that causes the page to get re-rendered at client boot up time (maybe [#338](https://github.com/angular/universal/issues/338#issuecomment-210493895) or [#359](https://github.com/angular/universal/issues/359)?)

**Comment by gdi2290** [comment]
yeah sorry, this is a problem with the recent change to include `ORIGIN_URL` where we're reusing the platform injectors but accidently reusing the application injectors as well

**Comment by gdi2290** [comment]
I'll push an update right now

**Comment by gdi2290** [comment]
@jhades  update to universal `0.98.1` fixes it

**Comment by jhades** [comment]
I've updated it and can confirm that everything is working well, Thx

**Comment by gdi2290** [comment]
make sure you're on beta.15

**Comment by vincentwinkel** [comment]
Hi,
I'm sorry but I've still the same problem. This is my package.json :
```
"angular2": "2.0.0-beta.15",
"angular2-universal": "~0.98.1",
```
You basic example works perfectly but when I try to insert my own code (another app I started to develop with angular2 starter), the client-side works perfectly, but the server always returns the index file containing empty <app></app>.
The build step is ok, everything build and run correctly, the problem is only about server-side.
I'll investigate but if you have an idea / suggestion to give me about that, let me know.
Btw, thank you for your work !

**Comment by gdi2290** [comment]
@vincentwinkel can you update universal to `0.98.2` then run `npm prune` and `npm install`

**Comment by visikov** [comment]
@gdi2290 I'm getting this issue with ng2-rc1 and ng2-uni@0.101.3

The seed project is producing the error as well, so you can pull that down to recreate the issue.

**Comment by ianbelcher** [comment]
Same issue here on ng2-rc1 and ng2-uni@0.101.3.

@visikov Did you figure out what your issue was? This is also happening for me another project I'm working on which leads me to believe the issue is with universal itself or I'm just not understanding how it is meant to be correctly implemented.

**Comment by visikov** [comment]
@ianbelcher I was only able to give it a few hours to investigate and couldn't find a solution. I'll try and take a look again sometime this weekend.

You're right, this is an issue with universal and not specific to this starter seed project.

**Comment by ishan123456789** [comment]
The issue still persists?
With Angular 6

**Comment by kalimulhaq** [comment]
Facing this issue in Angular 6
